plugins/badges-backend/src/lib/BadgeBuilder/DefaultBadgeBuilder.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Logger } from 'winston';
+import { makeBadge, ValidationError } from 'badge-maker';
+import { BadgeBuilder, BadgeConfig, BadgeOptions } from './types';
+import { interpolate } from '../../utils';
+
+export class DefaultBadgeBuilder implements BadgeBuilder {
+  constructor(
+    private readonly logger: Logger,
+    private readonly config: { [id: string]: BadgeConfig },
+  ) {}
+
+  public async getBadgeConfig(badgeId: string): BadgeConfig {
+    return (
+      this.config[badgeId] ||
+      this.config.default || {
+        label: 'Unknown badge ID',
+        message: badgeId,
+        color: 'red',
+      }
+    );
+  }
+
+  public async createBadge(options: BadgeOptions): string {
+    const { context, config: badge } = options;
+    const params = {
+      label: this.render(badge.label, context),
+      message: this.render(badge.message, context),
+      color: badge.color || '#36BAA2',
```

Comment:
should this be theme driven in some way? is there a color in the them that would be appropriate here, and thus if someone customized their backstage instance it would get picked up here also?

Reply:
That would be awesome. Will look into the theme..

Reply:
I think the way to go here is to provide the badge colors as query options when fetching them (i.e. in the badge url) as I don't have the theme in the backend. (to my knowledge.) 
I haven't looked into this yet, feeling a little bit like postponing it for a future update.